tests/unittests/test_datasource/test_azure.py

```diff
@@ -2270,6 +2418,220 @@ def test_reprovision_calls__poll_imds(self, _poll_imds, isfile):
         _poll_imds.assert_called_with()
 
 
+class TestPreprovisioningHotAttachNics(CiTestCase):
+
+    def setUp(self):
+        super(TestPreprovisioningHotAttachNics, self).setUp()
+        self.tmp = self.tmp_dir()
+        self.waagent_d = self.tmp_path('/var/lib/waagent', self.tmp)
+        self.paths = helpers.Paths({'cloud_dir': self.tmp})
+        dsaz.BUILTIN_DS_CONFIG['data_dir'] = self.waagent_d
+        self.paths = helpers.Paths({'cloud_dir': self.tmp})
+
+    @mock.patch('cloudinit.sources.helpers.netlink.wait_for_nic_detach_event')
+    @mock.patch(MOCKPATH + 'util.write_file')
```

Comment:
```suggestion
    @mock.patch(MOCKPATH + 'util.write_file', autospec=True)
```

I won't comment on every one of these that needs updating from now on: any mock on which we're calling an `assert_*` method should be autospec'd, per our HACKING doc: https://cloudinit.readthedocs.io/en/latest/topics/hacking.html#testing

Reply:
Thanks for the pointers. I used this opportunity to change the remaining places in the file as well that are missing autospec.